testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/model/UserSessionProviderTest.java

```diff
@@ -0,0 +1,762 @@
+ */
+public class UserSessionProviderTest extends AbstractTestRealmKeycloakTest {
+
+    @Deployment
+    @TargetsContainer(AUTH_SERVER_CURRENT)
+    public static WebArchive deploy() {
+        return RunOnServerDeployment.create(UserResource.class, UserSessionProviderTest.class)
+                .addPackages(true,
+                        "org.keycloak.testsuite",
+                        "org.keycloak.testsuite.model");
+    }
+
+
+    public static void setupRealm(KeycloakSession session){
+        RealmModel realm = session.realms().getRealmByName("test");
+        UserModel user1 = session.users().addUser(realm, "user1");
+        user1.setEmail("user1@localhost");
+        UserModel user2 = session.users().addUser(realm, "user2");
+        user2.setEmail("user2@localhost");
+    }
+    @Before
+    public void before() {
+        testingClient.server().run(UserSessionProviderTest::before);
+    }
+    public static void before(KeycloakSession session) {
+        RealmModel realm = session.realms().getRealmByName("test");
+        realm = session.realms().getRealm("test");
+        session.users().addUser(realm, "user1").setEmail("user1@localhost");
+        session.users().addUser(realm, "user2").setEmail("user2@localhost");
+    }
+
+    @After
+    public void after() {
+        testingClient.server().run(UserSessionProviderTest::after);
+    }
+    public static void after(KeycloakSession session) {
+        RealmModel realm = session.realms().getRealmByName("test");
+        session.sessions().removeUserSessions(realm);
+        UserModel user1 = session.users().getUserByUsername("user1", realm);
+        UserModel user2 = session.users().getUserByUsername("user2", realm);
+
+        UserManager um = new UserManager(session);
+        if (user1 != null) {
+            um.removeUser(realm, user1);
+        }
+        if (user2 != null) {
+            um.removeUser(realm, user2);
+        }
+    }
+    @Test
+    public void testCreateSessions() {
+        testingClient.server().run(UserSessionProviderTest::testCreateSessions);
+    }
+
+
+    public static void testCreateSessions(KeycloakSession session) {
+        int started = Time.currentTime();
+        RealmModel realm = session.realms().getRealmByName("test");
+        UserSessionModel[] sessions = createSessions(session);
+
+        assertSession(session.sessions().getUserSession(realm, sessions[0].getId()), session.users().getUserByUsername("user1", realm), "127.0.0.1", started, started, "test-app", "third-party");
+        assertSession(session.sessions().getUserSession(realm, sessions[1].getId()), session.users().getUserByUsername("user1", realm), "127.0.0.2", started, started, "test-app");
+        assertSession(session.sessions().getUserSession(realm, sessions[2].getId()), session.users().getUserByUsername("user2", realm), "127.0.0.3", started, started, "test-app");
+    }
+
+    @Test
+    public void testUpdateSession() {
```

Comment:
As we discussed, I am not a fan of "doubled" methods (each test method having separate static method). Is it possible to use @ModelTest instead and pass KeycloakSession as an argument to the method? Same for other places?